fvm/src/call_manager/default.rs

```diff
@@ -142,9 +143,9 @@ where
             }
             return Err(sys_err.into());
         }
-        self.call_stack_depth += 1;
+        self.push_call_stack();
         let result = self.send_unchecked::<K>(from, to, method, params, value);
-        self.call_stack_depth -= 1;
+        self.pop_call_stack();
```

Comment:
Hm, I don't know, it allows the caller to break some of the invariants this way by calling it with any value in `set_exec_memory_bytes`. If this enveloping could be done inside the limiter, it would be more self contained, but I doubt that's possible with the borrowing rules.

Reply:
Uploaded a version that does both: adds `with_stack_frame` to the call manager to encapsulate the depth check and maintenance, and also adds `with_stack_frame` to `ExecMemory` to allow the limiter to encapsulate the snapshotting and restoring of the memory without allowing anyone else to set arbitrary memory values. 

Thanks for the suggestion, better to rule out any misuse even if it was only called from a single place.